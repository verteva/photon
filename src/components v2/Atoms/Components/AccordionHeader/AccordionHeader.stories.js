import PAccordionHeader from './AccordionHeader.vue';
import PCard from '../Card/Card.vue';

export default {
  title: 'V2/Atoms/Components/AccordionHeader',
  component: PAccordionHeader,
  argTypes: {
    fullWidth: {
      control: {
        type: 'boolean',
      },
    },
    complete: {
      control: {
        type: 'boolean',
      },
    },
    expanded: {
      control: {
        type: 'boolean',
      },
    },
    disabled: {
      control: {
        type: 'boolean',
      },
    },
    interactive: {
      control: {
        type: 'boolean',
      },
    },
    openArrows: {
      control: {
        type: 'boolean',
      },
    },
    openCloseIcons: {
      control: {
        type: 'object',
      },
    },
    section: {
      control: {
        type: 'text',
      },
    },
  },
  args: {
    fullWidth: false,
    complete: false,
    expanded: false,
    disabled: false,
    interactive: false,
    openArrows: false,
    openCloseIcons: ['ChevronDown', 'ChevronUp'],
    section: 'section',
  },
  parameters: {
    backgrounds: {
      default: 'white',
      values: [
        { name: 'white', value: '#ffffff' },
        { name: 'light', value: '#efefef' },
        { name: 'dark', value: '#444444' },
      ],
    },
  },
};

const Template = (args, { argTypes }) => ({
  components: { PAccordionHeader, PCard },
  props: Object.keys(argTypes),
  template: '<PCard no-padding><PAccordionHeader v-bind="$props"/></PCard>',
});

export const Default = Template.bind({});
Default.args = {
  section: 'header',
  openArrows: true,
};

export const Expanded = Template.bind({});
Expanded.args = {
  section: 'header',
  openArrows: true,
  expanded: true,
};

export const Completed = Template.bind({});
Completed.args = {
  section: 'header',
  openArrows: true,
  complete: true,
};

export const Interactive = Template.bind({});
Interactive.args = {
  section: 'Interactive header',
  openArrows: true,
  interactive: true,
};

export const Disabled = Template.bind({});
Disabled.args = {
  section: 'Disabled header',
  openArrows: true,
  disabled: true,
};

export const CustomIcons = Template.bind({});
CustomIcons.args = {
  section: 'Custom Icons',
};
