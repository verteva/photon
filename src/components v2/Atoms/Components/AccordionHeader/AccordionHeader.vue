<template>
  <div
    v-if="hasHeaderContent"
    ref="accordionHeader"
    :class="{
      disabled,
      'full-width': fullWidth,
      'ph-accordion-header-empty': !section && !$slots.default,
      'ph-accordion-header-has-content': section || $slots.default,
      interactive,
    }"
    :tabindex="interactive ? 0 : -1"
    class="ph-accordion-header acc-header"
    @click="$emit('click', $event)"
    @focus="$emit('focus', $event)"
    @blur="$emit('blur', $event)"
  >
    <div class="ph-accordion-header-content" data-observe="true">
      {{ section }}
      <slot />
    </div>
    <font-awesome-icon
      v-if="complete"
      :icon="['fal', 'check']"
      class="ph-accordion-checkmark fa-fw"
      data-testid="complete-checkmark"
    />
    <div
      v-if="openArrows"
      class="ph-accordion-header-icon"
      data-testid="openArrow"
    >
      <font-awesome-icon
        :icon="['fal', !expanded ? 'chevron-down' : 'chevron-up']"
        class="ph-accordion-header-chevron fa-fw"
        :class="{
          expanded,
        }"
      />
    </div>
    <div
      v-if="!openArrows && openCloseIcons.length === 2"
      class="ph-accordion-header-icon"
    >
      <font-awesome-icon
        :icon="['fal', !expanded ? openCloseIcons[0] : openCloseIcons[1]]"
        class="fa-fw"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { PropType } from 'vue';

export const props = {
  fullWidth: {
    type: Boolean as PropType<boolean>,
    default: true,
  },
  complete: {
    type: Boolean as PropType<boolean>,
    default: false,
  },
  expanded: {
    type: Boolean as PropType<boolean>,
    default: false,
  },
  openArrows: {
    type: Boolean as PropType<boolean>,
    default: true,
  },
  openCloseIcons: {
    type: Array as PropType<Array<string>>,
    default: () => [],
  },
  section: {
    type: String as PropType<string>,
    default: '',
  },
  interactive: {
    type: Boolean as PropType<boolean>,
    default: true,
  },
  disabled: {
    type: Boolean as PropType<boolean>,
    default: false,
  },
};
export default {
  computed: {
    hasHeaderContent() {
      return this.section || this.$slots.default;
    },
  },
};
</script>

<script lang="ts" setup>
defineProps(props);
</script>

<style lang="scss">
@import './src/assets/scss/main.scss';

.ph-accordion-header {
  display: flex;
  align-items: center;
  width: 100%;

  &.ph-accordion-header-empty {
    padding-bottom: 0 !important;

    .ph-accordion-checkmark,
    .ph-accordion-header-icon {
      display: none;
    }
  }

  &.interactive {
    cursor: pointer;
  }

  &.disabled {
    cursor: not-allowed;
  }

  &.full-width {
    width: 100%;
  }

  .ph-accordion-header-icon {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-left: auto;
    color: var(--sd-theme-fg-default);
  }

  .ph-accordion-checkmark {
    margin-left: 8px;
    color: var(--sd-theme-success-default);
  }
}

.ph-accordion-header-content {
  @include token-heading('h5');
  width: 100%;
  margin-top: 0 !important;
  margin-bottom: 0 !important;
}
</style>
