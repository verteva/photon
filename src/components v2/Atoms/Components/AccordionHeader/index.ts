export { default, props } from './AccordionHeader.vue';
