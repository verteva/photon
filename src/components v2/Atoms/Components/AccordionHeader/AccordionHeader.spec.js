import { render } from '@testing-library/vue';
import userEvent from '@testing-library/user-event';
import AccordionHeader from './AccordionHeader.vue';

describe('AccordionHeader.vue', () => {
  const section = 'This is Accordion Header text';

  it('default', async () => {
    const { container } = render(AccordionHeader);

    expect(container).toBeTruthy();
  });

  it('props: complete', async () => {
    const { getByTestId } = render(AccordionHeader, {
      props: {
        section,
        complete: true,
      },
    });
    const check = getByTestId('complete-checkmark');
    expect(check).toBeTruthy();
  });

  it('props: section', async () => {
    const { getByText } = render(AccordionHeader, {
      props: {
        section,
      },
    });
    const item = getByText(section);
    expect(item).toBeTruthy();
  });

  it('props: interactive', async () => {
    const { container } = render(AccordionHeader, {
      props: {
        interactive: true,
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('props: disabled', async () => {
    const { container } = render(AccordionHeader, {
      props: {
        disabled: true,
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('props: fullWidth', async () => {
    const { container } = render(AccordionHeader, {
      props: {
        fullWidth: true,
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('props: openArrows = false', async () => {
    const { queryAllByTestId } = render(AccordionHeader, {
      props: {
        openArrows: false,
      },
    });
    const openArrow = await queryAllByTestId('openArrow');
    expect(openArrow.length).toBeFalsy();
  });

  it('props: expanded = true', async () => {
    const { container } = render(AccordionHeader, {
      props: {
        expanded: true,
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('props: openArrows = false, openCloseIcons', async () => {
    const { container } = render(AccordionHeader, {
      props: {
        openArrows: false,

        openCloseIcons: ['ChevronDown', 'ChevronUp'],
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('props: openArrows = false, expanded, openCloseIcons', async () => {
    const { container } = render(AccordionHeader, {
      props: {
        openArrows: false,
        expanded: true,
        openCloseIcons: ['ChevronDown', 'ChevronUp'],
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('emit: click', async () => {
    const section = 'I am the section!';
    const { emitted, getByText } = render(AccordionHeader, {
      props: {
        interactive: true,
        section,
      },
    });
    const item = getByText(section);
    expect(emitted().click).toBeFalsy();

    await userEvent.click(item);
    expect(emitted().click).toBeTruthy();
  });

  it('emit: focus', async () => {
    const section = 'I am the section!';
    const { emitted, getByText } = render(AccordionHeader, {
      props: {
        interactive: true,
        section,
      },
    });

    const item = getByText(section);
    expect(emitted().focus).toBeFalsy();

    await userEvent.click(item);
    expect(emitted().focus).toBeTruthy();
  });

  it('emit: blur', async () => {
    const section = 'I am the section!';
    const { emitted, getByText } = render(AccordionHeader, {
      props: {
        interactive: true,
        section,
      },
    });

    const item = getByText(section);
    expect(emitted().blur).toBeFalsy();

    await userEvent.click(item);
    expect(emitted().blur).toBeFalsy();

    await userEvent.tab();
    expect(emitted().blur).toBeTruthy();
  });
});
