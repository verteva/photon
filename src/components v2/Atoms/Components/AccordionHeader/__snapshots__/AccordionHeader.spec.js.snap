// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionHeader.vue props: disabled 1`] = `
<div>
  <div
    class="ph-accordion-header acc-header disabled full-width interactive"
    tabindex="0"
  >
    
  
   
    <!---->
     
    <div
      class="ph-ml-auto"
      data-testid="openArrow"
    >
      <div
        class="ph-accordion-header-chevron"
      >
        <svg
          class="picon"
          color="inherit"
          height="12"
          version="1.1"
          viewBox="0 0 12 7"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            id="Experiments"
            stroke="none"
            stroke-width="1"
          >
            <g
              id=<-06_2_Desktop->-FINANCIALS---Employment
              transform="translate(-1097.000000, -422.000000)"
            >
              <rect
                fill="url(#linearGradient-1)"
                height="1395"
                id="Mask"
                width="1440"
                x="0"
                y="0"
              />
               
              <rect
                height="7"
                id="Rectangle"
                width="12"
                x="1097"
                y="422"
              />
               
              <polyline
                id="Path-13"
                points="1098 423 1103 428 1108 423"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`AccordionHeader.vue props: expanded = true 1`] = `
<div>
  <div
    class="ph-accordion-header acc-header full-width interactive"
    tabindex="0"
  >
    
  
   
    <!---->
     
    <div
      class="ph-ml-auto"
      data-testid="openArrow"
    >
      <div
        class="ph-accordion-header-chevron expanded"
      >
        <svg
          class="picon"
          color="inherit"
          height="12"
          version="1.1"
          viewBox="0 0 12 7"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            id="Experiments"
            stroke="none"
            stroke-width="1"
          >
            <g
              id=<-06_2_Desktop->-FINANCIALS---Employment
              transform="translate(-1097.000000, -422.000000)"
            >
              <rect
                fill="url(#linearGradient-1)"
                height="1395"
                id="Mask"
                width="1440"
                x="0"
                y="0"
              />
               
              <rect
                height="7"
                id="Rectangle"
                width="12"
                x="1097"
                y="422"
              />
               
              <polyline
                id="Path-13"
                points="1098 423 1103 428 1108 423"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`AccordionHeader.vue props: fullWidth 1`] = `
<div>
  <div
    class="ph-accordion-header acc-header full-width interactive"
    tabindex="0"
  >
    
  
   
    <!---->
     
    <div
      class="ph-ml-auto"
      data-testid="openArrow"
    >
      <div
        class="ph-accordion-header-chevron"
      >
        <svg
          class="picon"
          color="inherit"
          height="12"
          version="1.1"
          viewBox="0 0 12 7"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            id="Experiments"
            stroke="none"
            stroke-width="1"
          >
            <g
              id=<-06_2_Desktop->-FINANCIALS---Employment
              transform="translate(-1097.000000, -422.000000)"
            >
              <rect
                fill="url(#linearGradient-1)"
                height="1395"
                id="Mask"
                width="1440"
                x="0"
                y="0"
              />
               
              <rect
                height="7"
                id="Rectangle"
                width="12"
                x="1097"
                y="422"
              />
               
              <polyline
                id="Path-13"
                points="1098 423 1103 428 1108 423"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`AccordionHeader.vue props: interactive 1`] = `
<div>
  <div
    class="ph-accordion-header acc-header full-width interactive"
    tabindex="0"
  >
    
  
   
    <!---->
     
    <div
      class="ph-ml-auto"
      data-testid="openArrow"
    >
      <div
        class="ph-accordion-header-chevron"
      >
        <svg
          class="picon"
          color="inherit"
          height="12"
          version="1.1"
          viewBox="0 0 12 7"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            id="Experiments"
            stroke="none"
            stroke-width="1"
          >
            <g
              id=<-06_2_Desktop->-FINANCIALS---Employment
              transform="translate(-1097.000000, -422.000000)"
            >
              <rect
                fill="url(#linearGradient-1)"
                height="1395"
                id="Mask"
                width="1440"
                x="0"
                y="0"
              />
               
              <rect
                height="7"
                id="Rectangle"
                width="12"
                x="1097"
                y="422"
              />
               
              <polyline
                id="Path-13"
                points="1098 423 1103 428 1108 423"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`AccordionHeader.vue props: openArrows = false, expanded, openCloseIcons 1`] = `
<div>
  <div
    class="ph-accordion-header acc-header full-width interactive"
    tabindex="0"
  >
    
  
   
    <!---->
     
    <!---->
     
    <div
      class="ph-ml-auto"
    >
      <div>
        <svg
          class="picon"
          color="inherit"
          height="18"
          version="1.1"
          viewBox="0 0 12 7"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            id="Experiments"
            stroke-width="1"
            transform="rotate(180, 6, 3)"
            troke="none"
          >
            <g
              id=<-06_2_Desktop->-FINANCIALS---Employment
              transform="translate(-1097.000000, -422.000000)"
            >
              <rect
                fill="url(#linearGradient-1)"
                height="1395"
                id="Mask"
                width="1440"
                x="0"
                y="0"
              />
               
              <rect
                height="7"
                id="Rectangle"
                width="12"
                x="1097"
                y="422"
              />
               
              <polyline
                id="Path-13"
                points="1098 423 1103 428 1108 423"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`AccordionHeader.vue props: openArrows = false, openCloseIcons 1`] = `
<div>
  <div
    class="ph-accordion-header acc-header full-width interactive"
    tabindex="0"
  >
    
  
   
    <!---->
     
    <!---->
     
    <div
      class="ph-ml-auto"
    >
      <div>
        <svg
          class="picon"
          color="inherit"
          height="18"
          version="1.1"
          viewBox="0 0 12 7"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            id="Experiments"
            stroke="none"
            stroke-width="1"
          >
            <g
              id=<-06_2_Desktop->-FINANCIALS---Employment
              transform="translate(-1097.000000, -422.000000)"
            >
              <rect
                fill="url(#linearGradient-1)"
                height="1395"
                id="Mask"
                width="1440"
                x="0"
                y="0"
              />
               
              <rect
                height="7"
                id="Rectangle"
                width="12"
                x="1097"
                y="422"
              />
               
              <polyline
                id="Path-13"
                points="1098 423 1103 428 1108 423"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
